addon/mixins/adapter-fetch.js

```diff
@@ -73,15 +73,20 @@ export function headersToObject(headers) {
  * Helper function that translates the options passed to `jQuery.ajax` into a format that `fetch` expects.
  * @param {Object} options
  */
-export function mungOptionsForFetch(_options) {
+export function mungOptionsForFetch(_options, adapter) {
   const options = (assign || merge)({
     credentials: 'same-origin',
   }, _options);
 
+  let adapterHeaders = adapter.get('headers');
+  if (adapterHeaders) {
+    options.headers = assign({}, options.headers || {}, adapterHeaders);
```

Comment:
The problem is that `merge` only takes 2 arguments, while assign takes N. If I was to use `merge` i'd have to do `merge(merge(a, b), c)`.